subcommand/inject-connect/command.go

```diff
@@ -323,15 +342,77 @@ func (c *Command) Run(args []string) int {
 		TLSConfig: &tls.Config{GetCertificate: c.getCertificate},
 	}
 
-	c.UI.Info(fmt.Sprintf("Listening on %q...", c.flagListen))
-	if err := server.ListenAndServeTLS("", ""); err != nil {
-		c.UI.Error(fmt.Sprintf("Error listening: %s", err))
-		return 1
+	// channel used for health checks
+	// also check to see if we should enable TLS
+	var healthCh chan struct{}
+	tlsEnabled := os.Getenv("CONSUL_CACERT")
+	consulPort := "8500"
+	if tlsEnabled != "" {
+		consulPort = "8501"
 	}
 
+	if c.flagEnableHealthChecks {
+		go func() {
+			// TODO error handling to close this
+			c.UI.Info(fmt.Sprintf("Listening on %q...", c.flagListen))
+			if err := server.ListenAndServeTLS("", ""); err != nil {
+				c.UI.Error(fmt.Sprintf("Error listening: %s", err))
+			}
+		}()
```

Comment:
We probably need to send this a channel so that we can shutdown the server gracefully

Reply:
@kschoche I think you need to address this comment because you will have a leaky goroutine otherwise. This server will start but has no way of terminating when the main function exits.

Reply:
Oh actually, ignore me! I think I saw in the latest code that you are closing the server on interrupt.